src/bsoncxx/json.cpp

```diff
@@ -274,15 +274,16 @@ std::string to_json(types::value value) {
 
     json_visitor v(ss, false, 0);
 
-    switch ((int)value.type()) {
-#define BSONCXX_ENUM(name, val)            \
-    case val:                              \
-        v.visit_value(value.get_##name()); \
-        break;
+    if(value.type()) {
```

Comment:
You could default-construct a document::view and attempt to use it, then it would be invalid in a similar way.

I think that we have a larger decision to make overall here in how default-constructed "invalid" types (like this one, document::view, uninitialized clients, etc.) should behave.  I think all such types in the driver will currently just seg fault if you use them this way, but we have open tickets like [CXX-742](https://jira.mongodb.org/browse/CXX-742) requesting that we fail in a friendlier way.